### DISCUSSION TITLE
milvus 新增了 querynode 之后，集群会自动把分片或者副本迁移到新加入的 querynode 吗？

### DISCUSSION THREAD

**Discussion by ponponon** [milvus-io/milvus#26192] [Q&A and General discussion]
milvus 新增了 querynode 之后，集群会自动把分片或者副本迁移到新加入的 querynode 吗？

因为 elasticsearch 会，所以 milvus 这种相同设计思想的分布式系统，应该也会吧？

还是会，我新增了 querynode 之后，需要手动「再平衡」？

-----------

When a querynode is added to milvus, will the cluster automatically migrate the slice or replica to the newly added querynode?

Since elasticsearch does it, should milvus, which is a distributed system with the same design idea, do it too?

Or will I need to manually "rebalance" the querynode when I add one?

**Comment by ponponon** [comment]
关于 elasticsearch 集群对于新加入节点的自动再平衡，chatGPT 是这么回答我的



> elasticsearch 新增了节点之后，集群会自动把分片或者副本迁移到新加入的节点吗？
>
> 
>
> 是的，Elasticsearch 在集群中新增节点时会自动进行分片和副本的重新分配，以实现数据的平衡和高可用性。这个过程称为分片再平衡（Shard Rebalancing）。
>
> 当你向 Elasticsearch 集群添加新节点时，Elasticsearch 会根据集群的配置和状态来决定如何重新分配分片和副本。分片再平衡的目标是确保每个节点上的分片数量相对均衡，从而提高查询和写入的性能，并保障数据的高可用性。Elasticsearch 会尽量将分片和副本从负载较高的节点移动到负载较低的节点上。
>
> 在分片再平衡过程中，Elasticsearch 会考虑每个节点的硬件资源、负载情况和网络延迟等因素，以确保分片的均衡分布。需要注意的是，分片再平衡可能会导致集群中的一些网络和资源消耗，因此在集群规模变大或者发生变化时，你可能需要根据情况调整分片再平衡的参数。
>
> 总之，Elasticsearch 在新增节点时会自动进行分片和副本的重新分配，以优化集群的性能和可用性。但请注意，具体的行为可能受到你的集群配置、版本和设置的影响，建议在进行重要操作之前查阅最新的 Elasticsearch 文档。



milvus 也是这样的吗？

**Reply by yhmo** [reply]
Milvus也是一样的，只不过milvus的balance主要考虑的是内存用量合理分配，没有把网络延迟这些算进去。

**Comment by mybabyiscue** [comment]
Hello, author. I saw the question you raised, so I guess you have already completed the cluster setup. I'm a newbie. Due to limited abilities. It is impossible to build a milvus cluster alone, so if you have relevant installation tutorials, can you provide one to me? If it's convenient for you, that would be very grateful.
